File: UI/careconnect_admin/lib/models/search_objects/employee_search_object.dart
import 'package:careconnect_admin/models/search_objects/employee_additional_data.dart';

import 'package:json_annotation/json_annotation.dart';

part 'employee_search_object.g.dart';

@JsonSerializable(explicitToJson: true)
class EmployeeSearchObject {
  String? fts;
  String? firstNameGTE;
  String? lastNameGTE;
  String? email;
  String? jobTitle;
  DateTime? hireDateGTE;
  DateTime? hireDateLTE;
  bool? employed;
  int? page;
  String? sortBy;
  bool? sortAscending;
  EmployeeAdditionalData? additionalData;
  bool? includeTotalCount;

  EmployeeSearchObject({
    this.fts,
    this.firstNameGTE,
    this.lastNameGTE,
    this.email,
    this.jobTitle,
    this.hireDateGTE,
    this.hireDateLTE,
    this.employed,
    this.page,
    this.sortBy,
    this.sortAscending,
    this.additionalData,
    this.includeTotalCount,
  });

  factory EmployeeSearchObject.fromJson(Map<String, dynamic> json) =>
      _$EmployeeSearchObjectFromJson(json);

  Map<String, dynamic> toJson() => _$EmployeeSearchObjectToJson(this);
}

File: CareConnect/CareConnect.Models/SearchObjects/EmployeeSearchObject.cs
using System;
using System.Collections.Generic;
using System.Text;
using CareConnect.Models.SearchObjects;

namespace CareConnect.Models.SearchObjects
{
    public class EmployeeSearchObject : BaseSearchObject<EmployeeAdditionalData>
    {
        public string? FirstNameGTE { get; set; }

        public string? LastNameGTE { get; set; }

        public string? Email { get; set; }

        public string? JobTitle { get; set; }

        public bool? Employed { get; set; }

        public DateTime? HireDateGTE { get; set; }

        public DateTime? HireDateLTE { get; set; }
    }
}
